sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/BulkWriter.scala

```diff
@@ -1273,6 +1290,20 @@ private object BulkWriter {
       }
     }
 
+  val emitFailureHandlerForComplete: EmitFailureHandler =
+    (signalType, emitResult) => {
+      if (emitResult.equals(EmitResult.FAIL_NON_SERIALIZED)) {
+        log.logDebug(s"emitFailureHandlerForComplete - Signal: ${signalType.toString}, Result: ${emitResult.toString}")
+        true
+      } else if (emitResult.equals(EmitResult.FAIL_CANCELLED) || emitResult.equals(EmitResult.FAIL_TERMINATED)) {
+        log.logDebug(s"emitFailureHandlerForComplete - Already completed - Signal: ${signalType.toString}, Result: ${emitResult.toString}")
```

Comment:
maybe info/warning for this case?

Reply:
Fail_Cancelled? No - that is fine - it is a well known race condition - don't think it makes ense to spam the logs with it